compare_rpm_repo.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+
+import argparse
+import bz2
+import json
+import lzma
+import requests
+import sqlite3
+import subprocess
+import tempfile
+import xml.etree.ElementTree as ET
+from collections import defaultdict
+from pathlib import Path
+
+from packaging.version import Version
+from list_updatable_packages import Spec
+
+DISTS = ['el8']
+COMPONENTS = [path.name for path in Path('packages/').glob('*')]
+NIGHTLY_PACKAGES = ['foreman', 'foreman-installer', 'foreman-proxy', 'foreman-release', 'foreman-selinux', 'rubygem-hammer_cli', 'rubygem-hammer_cli_foreman', 'rubygem-katello', 'katello-repos', 'rubygem-hammer_cli_katello', 'katello']
+
+
+def get_repo_packages(component, release='nightly', dist='el8', arch='x86_64', staging=False):
+    packages = defaultdict(lambda: '0')
+    if staging:
+        if component == 'plugins':
+            component = 'foreman-plugins'
+        elif component == 'client':
+            component = 'foreman-client'
+        if component == 'katello':
+            repo_url = f'http://koji.katello.org/releases/yum/{component}-{release}/{component}/{dist}/{arch}/'
+        else:
+            repo_url = f'http://koji.katello.org/releases/yum/{component}-{release}/{dist}/{arch}/'
+    else:
+        if component == 'foreman':
+            component = 'releases'
+        if component == 'katello':
+            repo_url = f'https://yum.theforeman.org/{component}/{release}/{component}/{dist}/{arch}/'
+        else:
+            repo_url = f'https://yum.theforeman.org/{component}/{release}/{dist}/{arch}/'
+    r = requests.get(f'{repo_url}/repodata/repomd.xml')
+    repomd = ET.fromstring(r.content)
+
+    sqlite_location = repomd.find('.//*[@type="primary_db"]/{http://linux.duke.edu/metadata/repo}location').get('href')
```

Comment:
FYI: https://fedoraproject.org/wiki/Changes/createrepo_c_1.0.0 has:

> Update createrepo_c to 1.0.0, new release will include change of default compression to zstd, no longer generating metadata in sqlite database format by default and simplified comps xml type in repodata. 

So that would invalidate this approach.

Reply:
Whyyyyyy. SQLite is so nice for that!

Reply:
Technically we can make it explicit and use `--database` to generate it, which is probably a good thing to do today. It mentions Fedora already does this.

It would be good to also look at zstd decompression.

You could also start to depend on https://pypi.org/project/repomd but that only has gzip support and also relies on defusedxml as an additional dependency.

Reply:
At that point I could just use `dnf` ;)

Doesn't zstd break EL7 which we still serve?

Reply:
Yes, but at this point it's just the defaults that they're proposing to change. It'll be a while before that affects us, but I just saw it pop up on fedora-devel and it looked relevant :)